src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            Operation operation = mapToOperation(parts[0]);
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
+        }
+        return transactions;
+    }
+
+    private Operation mapToOperation(String code) {
+        return switch (code) {
```

Comment:
Using the `switch` statement here contradicts the checklist's advice to avoid `switch-cases` and `if-else` constructions. Consider using a `Map<String, Operation>` to map operation codes to their corresponding `Operation` enum values.